homeassistant/components/huawei_lte/config_flow.py

```diff
@@ -89,6 +90,70 @@ async def _async_show_user_form(
             errors=errors or {},
         )
 
+    async def _async_show_reauth_form(
+        self,
+        user_input: dict[str, Any],
+        errors: dict[str, str] | None = None,
+    ) -> FlowResult:
+        return self.async_show_form(
+            step_id="reauth_confirm",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_USERNAME, default=user_input.get(CONF_USERNAME) or ""
+                    ): str,
+                    vol.Optional(
+                        CONF_PASSWORD, default=user_input.get(CONF_PASSWORD) or ""
+                    ): str,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    async def _try_connect(
+        self, user_input: dict[str, Any], errors: dict[str, str]
+    ) -> Connection | None:
+        """Try connecting with given data."""
+        username = user_input.get(CONF_USERNAME) or ""
+        password = user_input.get(CONF_PASSWORD) or ""
+
+        def _get_connection() -> Connection:
+            return Connection(
+                url=user_input[CONF_URL],
+                username=username,
+                password=password,
+                timeout=CONNECTION_TIMEOUT,
+            )
+
+        conn = None
+        try:
+            conn = await self.hass.async_add_executor_job(_get_connection)
+        except LoginErrorUsernameWrongException:
+            errors[CONF_USERNAME] = "incorrect_username"
+        except LoginErrorPasswordWrongException:
+            errors[CONF_PASSWORD] = "incorrect_password"
+        except LoginErrorUsernamePasswordWrongException:
+            errors[CONF_USERNAME] = "invalid_auth"
+        except LoginErrorUsernamePasswordOverrunException:
+            errors["base"] = "login_attempts_exceeded"
+        except ResponseErrorException:
+            _LOGGER.warning("Response error", exc_info=True)
+            errors["base"] = "response_error"
+        except Timeout:
+            _LOGGER.warning("Connection timeout", exc_info=True)
+            errors[CONF_URL] = "connection_timeout"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.warning("Unknown error connecting to device", exc_info=True)
+            errors[CONF_URL] = "unknown"
+        return conn
+
+    @staticmethod
+    def _logout(conn: Connection) -> None:
+        try:
+            conn.user_session.user.logout()  # type: ignore[union-attr]
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.debug("Could not logout", exc_info=True)
```

Comment:
Why not simply let the potential exceptions to bubble up here? This will swallow them for anyone who doesn't have enabled debug logger, making logs less useful.

Reply:
I don't think a successful logout is an essential part of a successful config flow. In that sense, we shouldn't make the flow fail even if it fails. And it arguably isn't something to bother users about in the usual case, even in form of logging it. Debug is there to make it possible to diagnose without code changes in case we happen to run into a case where this matters.

Unless we have a case at hand that shows this behavior is actually wrong or suboptimal, I think we shouldn't change it here -- this is what we've been doing for normal config flows already, with no problems reported that I'm aware of.

Reply:
Fair enough, that sounds good to me :+1: